disperser/common/blobstore/shared_storage.go

```diff
@@ -161,14 +161,28 @@ func (s *SharedBlobStore) MarkBlobDispersing(ctx context.Context, metadataKey di
 }
 
 func (s *SharedBlobStore) MarkBlobInsufficientSignatures(ctx context.Context, existingMetadata *disperser.BlobMetadata, confirmationInfo *disperser.ConfirmationInfo) (*disperser.BlobMetadata, error) {
+	if existingMetadata == nil {
+		return nil, errors.New("metadata is nil")
+	}
 	newMetadata := *existingMetadata
 	newMetadata.BlobStatus = disperser.InsufficientSignatures
-	newMetadata.ConfirmationInfo = confirmationInfo
+	if confirmationInfo != nil {
+		newMetadata.ConfirmationInfo = confirmationInfo
+	}
 	return &newMetadata, s.blobMetadataStore.UpdateBlobMetadata(ctx, existingMetadata.GetBlobKey(), &newMetadata)
 }
 
-func (s *SharedBlobStore) MarkBlobFinalized(ctx context.Context, metadataKey disperser.BlobKey) error {
-	return s.blobMetadataStore.SetBlobStatus(ctx, metadataKey, disperser.Finalized)
+func (s *SharedBlobStore) MarkBlobFinalized(ctx context.Context, existingMetadata *disperser.BlobMetadata, confirmationBlockNumber uint64) (*disperser.BlobMetadata, error) {
```

Comment:
It looks a more robust and simpler interface if the entire read-validate-update status-write back flow is in one function, as these steps are all tightly coupled.

Reply:
what do you mean by read-validate-update status-write back flow?

Reply:
Instead of taking a "existingMetadata" from the caller, this method can just take the potentially new confirmation block number: and internally it reads the metadata, compare the confirmation number with the given one, and if they are different, update it and write back. It's cleaner as it doesn't need to assume a "existingMetadata".

Reply:
I reverted this method back so it just updates the blob status. 
I added a separate method `UpdateConfirmationBlockNumber` and update the confirmation block number regardless of whether the block is finalized or not as we need to update the confirmation block even when it's in CONFIRMED state.